common/short-hex-str/src/lib.rs

```diff
@@ -81,22 +85,32 @@ const HEX_CHARS_LOWER: &[u8; 16] = b"0123456789abcdef";
 /// Format a byte as hex. Returns a tuple containing the first character and then
 /// the second character as ASCII bytes.
 #[inline(always)]
-fn byte2hex(byte: u8) -> (u8, u8) {
-    let hi = HEX_CHARS_LOWER[((byte >> 4) & 0x0f) as usize];
+fn byte2hex(byte: u8) -> Result<(u8, u8), Error> {
+    let hi = HEX_CHARS_LOWER[(byte
+        .checked_shr(4)
+        .ok_or_else(|| Error::OverflowError(String::from("byte2hex::byte")))?
+        & 0x0f) as usize];
     let lo = HEX_CHARS_LOWER[(byte & 0x0f) as usize];
-    (hi, lo)
+    Ok((hi, lo))
 }
 
 /// Hex encode a byte slice into the destination byte slice.
 #[inline(always)]
-fn hex_encode(src: &[u8], dst: &mut [u8]) {
-    debug_checked_precondition!(dst.len() == 2 * src.len());
+fn hex_encode(src: &[u8], dst: &mut [u8]) -> Result<(), Error> {
+    debug_checked_precondition!(
```

Comment:
The formatting/line-wrapping here looks interesting.. Have you run the auto formatter?

Something like `cargo x check` should work (it's an internal tool that wraps the rust auto formatters). I'll also run canary on the PR to see if the tests pass with these changes 😄